src/ol/source/wmtssource.js

```diff
@@ -428,6 +401,19 @@ ol.source.WMTS.optionsFromCapabilities = function(wmtsCap, config) {
         /urn:ogc:def:crs:(\w+):(.*:)?(\w+)$/, '$1:$3'));
   }
 
+  var projectionExtent = projection.getExtent();
+  var extent;
+  if (!goog.isNull(projectionExtent)) {
```

Comment:
but the spec doesn't require layers to have a bbox in the data projection (or a wgs84 one for that matter). http://www.ngi.be/cartoweb/1.0.0/WMTSCapabilities.xml is a simple example with layers in local projection and 3857. This is the Belgian mapping agency, so extent is limited to the area around Belgium.